tests/playwright-test/esm.spec.ts

```diff
@@ -135,13 +137,22 @@ test('should use source maps w/ ESM', async ({ runInlineTest, nodeVersion }) =>
       const { test } = pwt;
 
       test('check project name', ({}, testInfo) => {
+        expect(1).toBe(2);
         expect(testInfo.project.name).toBe('foo');
       });
     `
   }, { reporter: 'list' });
 
   const output = stripAnsi(result.output);
-  expect(result.exitCode).toBe(0);
-  expect(result.passed).toBe(1);
-  expect(output).toContain('a.test.ts:7:7');
+  expect(result.exitCode).toBe(1);
+  expect(result.failed).toBe(1);
+  expect.soft(output, 'error carrot—via source maps—is positioned appropriately').toContain(
+      [
+        `    >  8 |         expect(1).toBe(2);`,
+        `         |                   ^`
+      ].join('\n'));
```

Comment:
The above output is what we get if we don't have `"type": "module"`, so—assuming source map support is working—I'd expect to see the same output. However, with `"type": "module"` I just see:

```
  1) [foo] › a.test.ts:7:7 › check project name ====================================================

    Error: expect(received).toBe(expected) // Object.is equality

    Expected: 2
    Received: 1

        at file:///Users/rosswollman/code/microsoft/playwright/test-results/esm-should-use-source-maps-w-ESM-playwright-test/a.test.ts:8:19
```

i.e. no code frame indicating some misalignment with source mappings. This can be observed on both macOS and Windows.

Reply:
^ cc: @pavelfeldman What's expected behavior here? Is this a valid test?